oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/authentication/OidcIdTokenValidator.java

```diff
@@ -127,10 +112,51 @@ public OAuth2TokenValidatorResult validate(Jwt idToken) {
 		// The precise method for detecting replay attacks is Client specific.
 		// TODO Depends on gh-4442
 
+		if (!invalidClaims.isEmpty()) {
+			return OAuth2TokenValidatorResult.failure(invalidIdToken(invalidClaims));
+		}
+
 		return OAuth2TokenValidatorResult.success();
 	}
 
-	private static OAuth2TokenValidatorResult invalidIdToken() {
-		return OAuth2TokenValidatorResult.failure(INVALID_ID_TOKEN_ERROR);
+	private static OAuth2Error invalidIdToken(Map<String, Object> invalidClaims) {
```

Comment:
Let's remove the methods `requiredClaimError` and `invalidIdTokenError` and build the `OAuth2Error` in this method. Please ensure the message is formatted as mentioned in this [comment](https://github.com/spring-projects/spring-security/pull/6349/files#r246077923).

Reply:
Also, can you update the tests to assert on the `OAuth2Error.description`. And add a new that has at least 2 invalid claims present. Thanks!

Reply:
Ok, I've added the changes :)